endpoint_info.go

```diff
@@ -128,51 +127,42 @@ func (ep *endpoint) DriverInfo() (map[string]interface{}, error) {
 	return driver.EndpointOperInfo(nid, epid)
 }
 
-func (ep *endpoint) InterfaceList() []InterfaceInfo {
+func (ep *endpoint) Iface() InterfaceInfo {
 	ep.Lock()
 	defer ep.Unlock()
 
-	iList := make([]InterfaceInfo, len(ep.iFaces))
-
-	for i, iface := range ep.iFaces {
-		iList[i] = iface
+	if ep.iface != nil {
+		return ep.iface
 	}
 
-	return iList
+	return nil
 }
 
-func (ep *endpoint) Interfaces() []driverapi.InterfaceInfo {
+func (ep *endpoint) Interface() driverapi.InterfaceInfo {
```

Comment:
I know that issue.
In fact please note that `endpointInterface` is a structure not an interface.
What I was suggesting is just to change both functions' implementation to:

```
funcX() {
    if i := ep.getInterface(); i != nil {
        return i
    }
    return nil
}
```
The utility function will handle the locks and there are plenty of places in the code where it can be reused.